applications/ParticleMechanicsApplication/custom_strategies/strategies/mpm_residual_based_newton_raphson_strategy.hpp

```diff
@@ -228,7 +228,6 @@ class MPMResidualBasedNewtonRaphsonStrategy
         if (is_converged == true)
         {
             // Initialisation of the convergence criteria
-            r_dof_set = p_builder_and_solver->GetDofSet();
             this->mpConvergenceCriteria->InitializeSolutionStep(BaseType::GetModelPart(), r_dof_set, rA, rDx, rb);
```

Comment:
@philbucher Only solveSolutionStep() is re-implemented in the MPM strategy now. After some searching, the critical difference between the core solveSolutionStep and the MPM solveSolutionStep is this highlighted line, which occurs before the NL iteration loop. In the MPM it is present, but in the core NR it is not. 

By including this line, the MPM always converges in 2 iterations for my problem. 
![image](https://user-images.githubusercontent.com/23666229/80045900-2ef46a00-854c-11ea-9bc5-510e65ba7269.png)


By excluding it (as per the core), the MPM iterates 3 times - this is the cause of the performance difference.
![image](https://user-images.githubusercontent.com/23666229/80046228-61529700-854d-11ea-9424-e3aa70d19233.png)

The residual criterion is the convergence criterion btw. I'm not exactly sure what is happening here, but perhaps calling mpConvergenceCriteria->InitializeSolutionStep() pre-computes the residual norm, so that a solver iteration isn't wasted on computing it? Any ideas @tteschemacher , @RiccardoRossi ?